### ISSUE TITLE
[enhancement] Multi-input groups

### ISSUE THREAD

**Issue by dekrain** [jmoenig/Snap--Build-Your-Own-Blocks#1519]
For now, you can add multi input but with one fragment. I'd like to be possible add groups of multi-fragment. For example:
```
if %b %c
%mlt{else if %b %c}
else %c
```
to be:
```
if <> {
} [else if <> {
}]< > else
```

**Comment by cycomachead** [comment]
Yup! It should be on GitHub somewhere. I've wanted this too. 

And this is precisely the reason, though there are a few others.

**Comment by dekrain** [comment]
I'm working on it right now.

**Comment by brianharvey** [comment]
> I'm working on it right now.

Don't start writing code until we settle the hard design question, which is the user interface.  How do users make a group?  And what does the finished block look like?  Your picture shows a group with some text and a Boolean, but no C-slot.  Also, there's no reason for a new color (red) for this; we already have zebra coloring that we can use.

**Comment by dekrain** [comment]
For now I did this:
- CustomBlockDefinition.parseSpec - Parse multi-group input
- MultiArgMorph - Attribute `@isInline` for groups
- MultiArgMorph.getSpec - `%mult{...}`
- CustomBlockDefinition.parseSpec:%mult{...} -> BlockMorph.parseSpec - Now it's failing :/

**Comment by cycomachead** [comment]
The problem isn't the code, but these. There may be more, than what's here, but these are the current issues. They definitely aren't all thought out, so I imagine it will be a while until this is fully solved.


* "Make a block" screen: How do you designate a group? Adding removing inputs etc.
     * My guess is that we'd start off with some sub-dialog box. Like a Grouped Input option which includes it's own block input editor.
* how is the block prototype is drawn?
* How do you access the items in groups? I image they would be lists of lists, but this is TBD.
* Can input groups contain input groups?
    * It's totally possible to write a recursive parser for block specs that does the right thing...that doesn't necessarily mean it's a good idea.

**Comment by dekrain** [comment]
It's very new feature. I'm doing technical side for now.

**Comment by dekrain** [comment]
> Can input groups contain input groups?

Yes.

> How do you access the items in groups? I image they would be lists of lists, but this is TBD.

For example block `object with attribs %'attribs'{%s : %s}`:
```
arguments:scope
    attribs:list
        $element:list
            1:JSname // key (name: string, symbol)
            2:any // value
```

**Comment by brianharvey** [comment]
@dekrain:  When we say "don't write code until these issues are settled" it _doesn't_ mean you should unilaterally settle the issues by writing code.  It means, _**don't write code** until these issues are settled!_  There is absolutely **zero** chance that any code you write in a situation like this will actually be accepted.

You don't get to make design decisions.  Jens makes them, with input from the rest of us.  If you want to post pictures of what you think the issues Michael and I listed should look like _to the user,_ that might be helpful.  Code that you have specifically been told not to write is _not_ helpful.  Get it?

**Comment by brianharvey** [comment]
Like this:
![cond](https://cloud.githubusercontent.com/assets/3885074/20456124/ee25007e-ae3b-11e6-98a0-be428445f741.png)
except that the "else" isn't exactly where I'd want it to be.

**Comment by dekrain** [comment]
Yes, I know it. I wrote what I **did**. I did it before your first comment.

> Like this

It's looks fine.

**Comment by dekrain** [comment]
I know my code doesn't accepted, but my version of multi inputs: !!!
![image](https://cloud.githubusercontent.com/assets/19158291/20457541/210372bc-ae8e-11e6-9321-4f804cb61298.png)
Spec: `if %b %c %mult{else if %b %c} else %c`
